KlarnaOnDemand/KODLocalization.m

```diff
@@ -1,17 +1,28 @@
 #import "KODLocalization.h"
+#import "KODContext.h"
 
 @implementation KODLocalization
 
 + (NSBundle*)KODBundle {
   static NSBundle *bundle = nil;
+  static NSString *locale = nil;
   
-  if (!bundle) {
+  if (!bundle || locale != [KODContext getPreferredLocale]) {
+    locale = [KODContext getPreferredLocale];
     NSString *bundlePath = [NSBundle.mainBundle pathForResource:@"KOD" ofType:@"bundle"];
     if (!bundlePath) {
       bundlePath = [[NSBundle bundleForClass:[self class]] pathForResource:@"KOD" ofType:@"bundle"];
     }
     bundle = [NSBundle bundleWithPath:bundlePath];
+    
+    if (locale != [[NSBundle mainBundle] preferredLocalizations].firstObject) {
+      bundlePath = [bundle pathForResource:@"localization" ofType:@"strings" inDirectory:nil forLocalization:locale];
+      if (bundlePath) {
+        bundle = [NSBundle bundleWithPath:[bundlePath stringByDeletingLastPathComponent]];
+      }
+    }
   }
+
   return bundle;
 }
 
```

Comment:
I see it's working, but it's weird that `NSLocalizedStringFromTableInBundle` doesn't actually localize the string now.